### ISSUE TITLE
Transport endpoint is not connected

### ISSUE THREAD

**Issue by dbartenstein** [s3fs-fuse/s3fs-fuse#2176]
#### Version of s3fs being used (`s3fs --version`)
V1.89 (commit:unknown)

#### Version of fuse being used (`pkg-config --modversion fuse`, `rpm -qi fuse` or `dpkg -s fuse`)
2.9.9-5

#### Kernel information (`uname -r`)
5.10.0-23-cloud-amd64

#### GNU/Linux Distribution, if applicable (`cat /etc/os-release`)
Debian GNU/Linux 11 (bullseye)

#### How to run s3fs, if applicable

[x] /etc/fstab  

```
s3fs#prod.media.zemtu   /var/www/zemtu/public/media     fuse    _netdev,allow_other,use_path_request_style,use_cache=/tmp,uid=1001,gid=1002,umask=0022,url=https://sos-ch-dk-2.exo.io/,passwd_file=/root/.passwd-s3fs,comment=cloudconfig  0       0
```

#### s3fs syslog messages (`grep s3fs /var/log/syslog`, `journalctl | grep s3fs`, or `s3fs outputs`)

```
syslog
[Errno 107] Transport endpoint is not connected

May 29 09:16:58 prod-main-zemtu s3fs[2017306]: s3fs: unable to access MOUNTPOINT /var/www/zemtu/public/media: Transport endpoint is not connected
```

### Details about issue
After two weeks without any issues, the mountpoint suddenly stopped working. We had to execute:
```
sudo fusermount -uz /var/www/zemtu/public/media
sudo mount -a
```

I already commented another similar issue: https://github.com/s3fs-fuse/s3fs-fuse/issues/725#issuecomment-1568047863
Hint: we’re using [Exoscale](https://www.exoscale.com/), not AWS.

An option I came across is `nodnscache (disable DNS cache)` - what exactly happens when the dnscache is enabled (default)? Will s3fs pick up DNS changes automatically, e.g. when the URL suddenly points to a CNAME instead of an A record?

Thanks a lot!

**Comment by dbartenstein** [comment]
Update: we have just upgraded to s3fs 1.92 and will continue monitoring the situation.

But we still wonder what` nodnscache (disable DNS cache)` exactly does. For how long is DNS cached when the option is not used? Thanks!

**Comment by ggtakec** [comment]
@dbartenstein Thanks for testing with the latest code(v1.92).
```
s3fs: unable to access MOUNTPOINT /var/www/zemtu/public/media: Transport endpoint is not connected
```
You are reporting this error in your logs, is this error preventing s3fs from starting?
If s3fs does not start due to the above error, it means that the mount point(`/var/www/zemtu/public/media`) could not be accessed by the s3fs effective user.

If the error above and the reason s3fs doesn't start are different, we need to await other log/report by your environment.
Thanks in advance for your assistance.

**Comment by dbartenstein** [comment]
> @dbartenstein Thanks for testing with the latest code(v1.92).
> 
> ```
> s3fs: unable to access MOUNTPOINT /var/www/zemtu/public/media: Transport endpoint is not connected
> ```
> 
> You are reporting this error in your logs, is this error preventing s3fs from starting? If s3fs does not start due to the above error, it means that the mount point(`/var/www/zemtu/public/media`) could not be accessed by the s3fs effective user.
> 
> If the error above and the reason s3fs doesn't start are different, we need to await other log/report by your environment. Thanks in advance for your assistance.

@ggtakec 
s3fs had been running for two weeks and then it suddenly stopped with the "Transport endpoint is not connected" error. Only after executing the following lines, the end point was back again:
```
sudo fusermount -uz /var/www/zemtu/public/media
sudo mount -a
```

> But we still wonder what nodnscache (disable DNS cache) exactly does. For how long is DNS cached when the option is not used? Thanks!

What exactly is the option `nodnscache` doing?

Thanks!

**Comment by gaul** [comment]
Please reopen if symptoms persist.

**Comment by uncdubdiver** [comment]
Can I request we reopen this? It's still happening to me constantly.
![image](https://github.com/s3fs-fuse/s3fs-fuse/assets/7240221/b279cf3b-2277-4184-98af-0131a09409a4)

**Comment by gaul** [comment]
This is a generic symptom that indicates a crash.  Please open a new issue for your symptom.
